src/azure.rs

```diff
@@ -303,6 +447,11 @@ pub fn new_azure(
     };
 
     let storage_client = storage_account_client.as_storage_client();
+    let blob_base_url = storage_account_client
+        .blob_storage_url()
+        .as_ref()
+        .trim_end_matches('/')
```

Comment:
the failing tests were relates to trailing slashes being inconsistent between azurite and normal. thought about using the url crate to avoid this, but then felt in this simple case string manipulation works as well.. let me know if you feel this should be changed though :).